### PR TITLE
TimePunch Dates Restriction

### PR DESCRIPTION
**Pull request by MunifNagi**
As a user, you shouldn't be able to:
-Request a future time punch
-Request a  vacation  with  vacation start date after vacation end date
As an admin, you shouldn’t be able to:
-Add a future event

### CODE REVIEW

**Comment by joelbcastillo** [comment]
@MunifNagi - Is there an issue associated with this?

**Comment by MunifNagi** [comment]
> @MunifNagi - Is there an issue associated with this?

no, I just thought a user/admin should receive an error message when adding a not valid TimePunch/Event (i.e future time-punch, past vacation).

File: app/main/views.py [hunk 1]
```diff
@@ -514,9 +524,15 @@ def request_timepunch():
             flash("You must have a supervisor to request a timepunch. If you believe a supervisor "
                   "should be assigned to you, please contact the system administrator.", category='error')
             current_app.logger.error('Does not have a supervisor'.format(current_user.email))
+        elif vacform.vac_start.data > vacform.vac_end.data:
```
Comment (joelbcastillo) [inline app/main/views.py]: There might be a valid reason to submit a vacation after the fact.

File: app/main/views.py [hunk 2]
```diff
@@ -514,6 +522,8 @@ def request_timepunch():
...
+            flash("The vacation start data must be before the vacation end date", category='error')
```
Comment (joelbcastillo) [inline app/main/views.py]: Typo here.

**Comment by joelbcastillo** [comment]
@MunifNagi  - Please resolve merge conflicts. Thanks

**Pull request merged**